app/MMB/src/navigation/AddressPickerScreen.js

```diff
@@ -95,23 +102,41 @@ export class AddressPickerScreen extends React.Component<Props, State> {
     }
   });
 
+  setLocationToCurrentLocation = () => {
+    const location = this.getCoordinate();
+    if (location != null) {
+      this.props.navigation.state.params.setLocation(location);
+    }
+  };
+
   render() {
+    const searchText =
+      idx(this.props.navigation, _ => _.state.params.searchText) || '';
+    const region = idx(this.props.navigation, _ => _.state.params.region);
     return (
-      <ScrollView>
-        {this.state.formattedAddress != '' && (
-          <View style={styles.currentLocationContainer}>
-            <TextIcon code="&quot;" style={styles.currentLocationIcon} />
-            <View>
-              <Text style={styles.currentLocationTitle}>
-                <Translation id="mmb.trip_service.transportation.address_picker.current_location_title" />
-              </Text>
-              <Text style={styles.currentLocationAddress}>
-                <Translation passThrough={this.state.formattedAddress} />
-              </Text>
+      <ScrollView keyboardShouldPersistTaps="handled">
```

Comment:
Keyboard should normally close on scroll, and also you should use Keyboard avoiding view.